### PR TITLE
Drop stored scripts with the old style-id

### PR DESCRIPTION
**Pull request by jdconrad**
This PR fixes (https://github.com/elastic/elasticsearch/issues/47593). Stored scripts with the old-style id of lang#id are saved through the upgrade process but are no longer accessible in recent versions. This fix will drop those scripts altogether since there is no way for a user to access them.

### CODE REVIEW

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/1

File: server/src/test/java/org/elasticsearch/script/ScriptMetaDataTests.java [hunk 1]
```diff
@@ -168,6 +171,41 @@ public void testLoadEmptyScripts() throws IOException {
...
+            builder.endObject();
+        }
+        builder.endObject();
+
+        XContentParser parser = XContentType.JSON.xContent()
+                .createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+                        BytesReference.bytes(builder).streamInput());
+        ScriptMetaData smd = ScriptMetaData.fromXContent(parser);
+        assertNull(smd.getStoredScript("painless#test"));
+        assertNull(smd.getStoredScript("lang#test"));
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/script/ScriptMetaDataTests.java]: Maybe check the size of the Map returned from `getStoredScripts()`?

**Review by rjernst — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/script/ScriptMetaDataTests.java (hunk 1 above)
Comment (jdconrad) [reply server/src/test/java/org/elasticsearch/script/ScriptMetaDataTests.java]: Good call. Added.

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/2

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by jdconrad** [comment]
@martijnvg @rjernst Thanks for the reviews! Will commit as soon as CI passes.

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**